src/main/scala/org/apache/spark/shuffle/ucx/perf/UcxPerfBenchmark.scala

```diff
@@ -107,21 +114,33 @@ object UcxPerfBenchmark extends App with Logging {
     val blocks = Array.ofDim[BlockId](options.numOutstanding)
     val callbacks = Array.ofDim[OperationCallback](options.numOutstanding)
     val requestInFlight = new AtomicInteger(0)
+    val rnd = new scala.util.Random
+    val blocksPerFile = options.numBlocks / options.files.length
+
+    val blockCollection =  if (options.numThreads > 1) {
+      val parallelCollection = (0 until options.numBlocks by options.numOutstanding).par
+      val threadPool = ThreadUtils.newForkJoinPool("Benchmark threads", options.numThreads)
+      parallelCollection.tasksupport = new ForkJoinTaskSupport(threadPool)
+      parallelCollection
+    } else {
+      (0 until options.numBlocks by options.numOutstanding)
+    }
 
     for (_ <- 0 until options.numIterations) {
-      for (b <- 0 until options.numBlocks by options.numOutstanding) {
+      for  (b <- blockCollection) {
         requestInFlight.set(options.numOutstanding)
         for (o <- 0 until options.numOutstanding) {
-          val fileIdx = (b+o) % options.files.size
-          val blockIdx = (b+o) / options.files.size
+          val fileIdx = if (options.randOrder) rnd.nextInt(options.files.length) else (b+o) % options.files.length
+          val blockIdx = if (options.randOrder) rnd.nextInt(blocksPerFile) else (b+o) % blocksPerFile
```

Comment:
after a second thought, i think it should use `(b+o) / options.files.length` instead of `(b+o) % blocksPerFile`. wdyt?

Reply:
Each file has total blocks / num files. E.g. total blocks is 1000, num files is 10 - so blocks per file is 10